### ISSUE TITLE
Terms and Conditions not appearing when logging in to the website

### ISSUE THREAD

**Issue by robin-hutchinson** [BiologicalRecordsCentre/iRecord#1346]
A user (339629) reported this issue:

"I have downloaded your app and verified my account all seemed to be well uploaded records etc but when I logon to the main site I just get taken to a T&C's section with another logon button.  Although the logon is accepted I don't seem to be properly logged in.  No records to view etc."

I tried to masquerade as their account, but when I click the masquerade button I am rerouted to this page:

![image](https://user-images.githubusercontent.com/95474755/187405639-310a0b86-4c74-4218-b27f-46ed3e8ce6b5.png)

And then when I click home I have to log back in to my own account. I have asked the user to clear the cookies in their browser, which did not work. They have also tried using different devices and browsers.

**Comment by BirenRathod** [comment]
@johnvanbreda For legal module, one of the requirement is for new user to accept first T & C once logged-in. If that is not fulfilled then obviously it will logged them out. One way to by pass this is to allow all "Authenticated" roles to expempt from it. That way it could stop user to logged them out. Not sure this is the right approach.

There is a setting for only new user to accept t&c and existing will be accepted by default. So when the new version of T&C comes out, you can change that to accept it by all.

**Comment by johnvanbreda** [comment]
@BirenRathod but the user is being logged out when the module tries to show them the T&Cs for them to accept. So they never get the chance to accept it. 

I think the legal module works OK if you register via the Drupal website. But, if you register via some other means (e.g. the app which uses the indicia api module to programmatically register the user) then the T&Cs can never be accepted.

**Comment by BirenRathod** [comment]
@johnvanbreda yes, that is what I explained that user registered via app needs to be accepted that T & C. so when next time they logged-in by website, it should show T & c too. There is a check-box in the legal settings which says show T & C everytime when logged-in. At present that is not enabled. The downside may be is that it could show everytime regardless of its accepted status. I think, we can use that option to avoid this scenario.

**Comment by kitenetter** [comment]
Is there still an issue to resolve here?

**Comment by johnvanbreda** [comment]
@kitenetter yes I don't think this has been fixed.

**Comment by kitenetter** [comment]
@johnvanbreda is this something that you need to work on, or Biren, or Karolis?

**Comment by johnvanbreda** [comment]
Probably myself or Biren, enlisting Karolis if required.
